fvtest/compilerunittest/CodeGenTest.hpp

```diff
@@ -36,6 +39,84 @@ class CodeGenTest : public TRTest::CompilerUnitTest {
     TR::Node *fakeNode;
 };
 
+class BinaryInstruction {
+public:
+    std::vector<uint32_t> _words;
+
+    BinaryInstruction() {}
+
+    BinaryInstruction(uint32_t instr) {
+        _words.push_back(instr);
+    }
+
+    BinaryInstruction(uint32_t prefix, uint32_t suffix) {
+        _words.push_back(prefix);
+        _words.push_back(suffix);
+    }
+
+    BinaryInstruction(std::vector<uint32_t> words) : _words(words) {}
+
+    const std::vector<uint32_t>& words() const { return _words; }
+
+    BinaryInstruction prepend(BinaryInstruction other) const {
+        std::vector<uint32_t> new_words(other._words);
+
+        for (size_t i = 0; i < _words.size(); i++)
+            new_words.push_back(_words[i]);
+
+        return new_words;
+    }
+
+    BinaryInstruction operator^(const BinaryInstruction& other) const {
+        if (_words.size() != other._words.size())
+            throw std::invalid_argument("Attempt to XOR instructions of different sizes");
+
+        BinaryInstruction instr(_words);
+
+        for (size_t i = 0; i < _words.size(); i++)
+            instr._words[i] ^= other._words[i];
+
+        return instr;
+    }
+
+    bool operator==(const BinaryInstruction& other) const {
+        return other._words == _words;
+    }
+};
+
+class BinaryEncoderTest : public TRTest::CodeGenTest {
+public:
+    uint32_t buf[INSTRUCTION_MAX_WORDS];
+
+    BinaryEncoderTest() {
+        cg()->setBinaryBufferStart(reinterpret_cast<uint8_t*>(&getAlignedBuf()[0]));
+    }
+
+    uint32_t* getAlignedBuf();
```

Comment:
I'm not a fan of implementing this method in multiple places to get platform-specific behaviour, espcially since the required size of `buf` depends on what alignment is required here. IMO a template argument to control alignment would be better.